python/data/dataset/coco_dataset.py

```diff
@@ -0,0 +1,69 @@
+
+import os
+import cv2
+import numpy as np
+from .dataset import MapDataset
+from ..preprocess import transform
+from ..utils import create_operators
+from pycocotools.coco import COCO
+
+class COCODataset(MapDataset):
+    def __init__(self,
+                 image_dir=None,
+                 annotation_path=None,
+                 transform_ops=None):
+        super().__init__()
+
+        self.image_dir = image_dir
+        self.annotation_path = annotation_path
+
+        if transform_ops:
+            self.transform_ops = create_operators(transform_ops)
```

Comment:
(1) self.transform_ops is not set when transfrom_ops=None;
(2) please use a different name if transform_ops changes its type.